### ISSUE TITLE
error in rule get_bins() in binning workflow

### ISSUE THREAD

**Issue by taylorreiter** [metagenome-atlas/atlas#224]
while running `atlas run binning`, I receive this error:

```
(atlas) stamps19@149.165.156.135:~/atlas_workflow$ atlas run binning
[2019-07-29 16:49 INFO] Executing: snakemake --snakefile /opt/miniconda3/envs/atlas/lib/python3.6/site-packages/atlas/Snakefile --directory /home/stamps19/atlas_workflow --jobs 6 --rerun-incomplete --configfile '/home/stamps19/atlas_workflow/config.yaml' --nolock  --use-conda  --conda-prefix /home/stamps19/atlas_workflow/databases/conda_envs   binning
Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 6
Rules claiming more threads will be scaled down.
Job counts:
        count   jobs
        1       binning
        1       build_bin_report
        1       get_bins
        1       run_checkm_lineage_wf
        1       run_checkm_tree_qa
        5

[Mon Jul 29 16:49:57 2019]
localrule get_bins:
    input: SRS049959/binning/DASTool/cluster_attribution.tsv, SRS049959/SRS049959_contigs.fasta
    output: SRS049959/binning/DASTool/bins
    jobid: 19
    wildcards: sample=SRS049959, binner=DASTool

Activating conda environment: /home/stamps19/atlas_workflow/databases/conda_envs/7dd678c1
Traceback (most recent call last):
  File "/opt/miniconda3/envs/atlas/lib/python3.6/site-packages/pandas/__init__.py", line 30, in <module>
    from pandas._libs import hashtable as _hashtable, lib as _lib, tslib as _tslib
  File "/opt/miniconda3/envs/atlas/lib/python3.6/site-packages/pandas/_libs/__init__.py", line 3, in <module>
    from .tslibs import (
  File "/opt/miniconda3/envs/atlas/lib/python3.6/site-packages/pandas/_libs/tslibs/__init__.py", line 3, in <module>
    from .conversion import localize_pydatetime, normalize_date
ModuleNotFoundError: No module named 'pandas._libs.tslibs.conversion'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/stamps19/atlas_workflow/.snakemake/scripts/tmp10v24n32.get_fasta_of_bins.py", line 10, in <module>
    import pandas as pd
  File "/opt/miniconda3/envs/atlas/lib/python3.6/site-packages/pandas/__init__.py", line 38, in <module>
    "the C extensions first.".format(module)
ImportError: C extension: No module named 'pandas._libs.tslibs.conversion' not built. If you want to import pandas from the source directory, you
 may need to run 'python setup.py build_ext --inplace --force' to build the C extensions first.
[Mon Jul 29 16:51:47 2019]
Error in rule get_bins:
    jobid: 19
    output: SRS049959/binning/DASTool/bins
    conda-env: /home/stamps19/atlas_workflow/databases/conda_envs/7dd678c1

RuleException:
CalledProcessError in line 294 of /opt/miniconda3/envs/atlas/lib/python3.6/site-packages/atlas/rules/binning.snakefile:
Command 'source /opt/miniconda3/bin/activate '/home/stamps19/atlas_workflow/databases/conda_envs/7dd678c1'; set -euo pipefail;  python /home/stam
ps19/atlas_workflow/.snakemake/scripts/tmp10v24n32.get_fasta_of_bins.py' returned non-zero exit status 1.
  File "/opt/miniconda3/envs/atlas/lib/python3.6/site-packages/atlas/rules/binning.snakefile", line 294, in __rule_get_bins
  File "/opt/miniconda3/envs/atlas/lib/python3.6/concurrent/futures/thread.py", line 56, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Note the path to the log file for debugging.
Documentation is available at: https://metagenome-atlas.readthedocs.io

Complete log: /home/stamps19/atlas_workflow/.snakemake/log/2019-07-29T164956.414631.snakemake.log
[2019-07-29 16:51 CRITICAL] Command 'snakemake --snakefile /opt/miniconda3/envs/atlas/lib/python3.6/site-packages/atlas/Snakefile --directory /ho
me/stamps19/atlas_workflow --jobs 6 --rerun-incomplete --configfile '/home/stamps19/atlas_workflow/config.yaml' --nolock  --use-conda  --conda-pr
efix /home/stamps19/atlas_workflow/databases/conda_envs   binning   ' returned non-zero exit status 1.
```

I tried bumping the versions of pandas and biopython in the environment and that was not successful in solving the problem. Any ideas as to what might be causing this?

**Comment by SilasK** [comment]
This has probably to do with the das tool bioconda recipie

**Comment by SilasK** [comment]
The error seems to be with the pandas version in the atlas environment. 
Atlas was built with pandas version 0.23, 
the bioconda recipie allows for pandas >0.23
maybe you can require this version by the installation of atlas. 

the new bioconda recipie is on the way.

**Comment by botellaflotante** [comment]
I also found an error in rule_get_bins, re-running binning on a sample that did not have this error previously... This is what I could find about it

Error in rule get_bins:
    jobid: 244
    output: SRR6797244/binning/DASTool/bins
    log: SRR6797244/logs/binning/get_bins_DASTool.log (check log file(s) for error message)
    conda-env: /linuxhome/tmp/jose/databases/conda_envs/984423c9

RuleException:
CalledProcessError in line 303 of /home/jose/miniconda3/envs/atlasenv/lib/python3.6/site-packages/atlas/rules/binning.snakefile:
Command 'source /home/jose/miniconda3/bin/activate '/linuxhome/tmp/jose/databases/conda_envs/984423c9'; set -euo pipefail;  python /linuxhome/tmp/jose/wdir_coumarins/.snakemake/scripts/tmpg2ksfql9.get_fasta_of_bins.py' returned non-zero exit status 1.
  File "/home/jose/miniconda3/envs/atlasenv/lib/python3.6/site-packages/atlas/rules/binning.snakefile", line 303, in __rule_get_bins
  File "/home/jose/miniconda3/envs/atlasenv/lib/python3.6/concurrent/futures/thread.py", line 56, in run
Removing temporary output file SRR6797250/binning/coverage/SRR6797250_coverage.txt.

SRR6797244/logs/binning/get_bins_DASTool.log seems to be empty...

any clue? thanks!

**Comment by SilasK** [comment]
I should have implemented the logging.. 

Can you check the input file to this rule: "{sample}/binning/DASTool/cluster_attribution.tsv"
and also the log file in {sample}/binning/DASTool

**Comment by botellaflotante** [comment]
yes, so they seem to be many bins in the first and many bins selected in the second:
bin selection complete: 32 bins above score threshold selected. 
here's the log
[SRR6797244_DASTool.log](https://github.com/metagenome-atlas/atlas/files/5815870/SRR6797244_DASTool.log)
 many thanks

**Comment by SilasK** [comment]
Ok you should have bins.

Then the file "{sample}/binning/DASTool/cluster_attribution.tsv" should look like:
something like.
cintig_1.   bin_nr
contig_4.  bin_nr


If this is the case run atlas again and capture the full log.

`atlas run binning options &> atlas.log`

If the get_bins fils it should print an error somewhere in the stdout/or stderr captured in the atlas.log

I'm sorry that the logging doesn't work properly.

**Comment by botellaflotante** [comment]
ok thanks! I think it is a problem with numpy...
[atlas.log](https://github.com/metagenome-atlas/atlas/files/5817638/atlas.log)
thanks a lot

**Comment by SilasK** [comment]
Thank you for finding it out. I think it has to be that conda installed python 3.8. I try to fix this in #357 but I would need some testing. 

You should be able to fix this on your side with:

```
conda activate /linuxhome/tmp/jose/databases/conda_envs/984423c9
conda install -y mamba
mamba install -y python3.7  biopython=1
conda deactivate
```
